src/query/storages/fuse/src/statistics/column_statistic.rs

```diff
@@ -74,71 +74,91 @@ pub fn gen_columns_statistics(
             continue;
         }
 
-        // later, during the evaluation of expressions, name of field does not matter
-        let mut min = Scalar::Null;
-        let mut max = Scalar::Null;
-
-        let (mins, _) = eval_aggr("min", vec![], &[col.clone()], rows)?;
-        let (maxs, _) = eval_aggr("max", vec![], &[col.clone()], rows)?;
-
-        if mins.len() > 0 {
-            min = if let Some(v) = mins.index(0) {
-                if let Some(v) = v.to_owned().trim_min(STATS_STRING_PREFIX_LEN) {
-                    v
+        match col {
+            Value::Scalar(s) => {
+                let (distinct_of_values, unset_bits) = if *s == Scalar::Null {
+                    (0, rows)
                 } else {
-                    continue;
-                }
-            } else {
-                continue;
+                    (1, 0)
+                };
+                let col_stats = ColumnStatistics::new(
+                    s.clone(),
+                    s.clone(),
+                    unset_bits as u64,
+                    0,
```

Comment:
the scalar value will be converted to a column when write to parquet file, so the `in_memory_size` should be the memory size of scalar value multiply by `rows`.

Reply:
the field is called in_memory_size, so not need to  multiply by rows?

Reply:
fixed